### ISSUE TITLE
Improving implementation of maps

### ISSUE THREAD

**Issue by chiragw15** [fossasia/susi_android#652]
**Actual Behaviour**

```
  "actions": [
      {
        "type": "answer",
        "expression": "Singapore is a place with a population of 3547809. Here is a map: https://www.openstreetmap.org/#map=13/1.2896698812440377/103.85006683126556"
      },
      {
        "type": "anchor",
        "link": "https://www.openstreetmap.org/#map=13/1.2896698812440377/103.85006683126556",
        "text": "Link to Openstreetmap: Singapore"
      },
      {
        "type": "map",
        "latitude": "1.2896698812440377",
        "longitude": "103.85006683126556",
        "zoom": "13"
      }
    ]
```
This is response from server. We aren't using data provided along with the answer but we are extracting data from link given in the answer.

This is wrong because expression of answer may change.

**Expected Behaviour**

Use "latitude" , "longitude" and "zoom" given in type "map" for getting map image from.
`"http://staticmap.openstreetmap.de/staticmap.php?center=LATITUDE,LONGITUDE&zoom=ZOOM&size=300x300`
just replace them in the link.

If user has google maps installed: 
Use the latitude and longitude from "map" for intent data when user clicks on map image.
If user doesn't have google maps installed:
Also use link in the "anchor" and open that link when user clicks on map image.  

@mayank408 @amitiwary999 Anyone who is free, will pick up an issue, assign the issue to himself and assign milestone on the issue and will start working on it. Always assign issues to yourself. By this way we will know, who is working on which issue. Same applies to every issue.

**Comment by amitiwary999** [comment]
@chiragw15 I want to work on this issue. But as I mention in my scrum I will start working from Monday because my laptop is not working.

**Comment by chiragw15** [comment]
Alright, I am assigning this to you and adding milestone 5.

**Comment by chiragw15** [comment]
Fixed in #681
